### PR TITLE
refs #8874 - updates for gettext 3

### PR DESCRIPTION
**Pull request by orrabin**
I added the changes from: https://github.com/theforeman/foreman_plugin_template/pull/13 and tried it here with https://github.com/theforeman/foreman/pull/2096.
I did @bkearney's 3 steps from the discussion on the core pr and they worked.
Then I changed an extracted string and ran step a again (`rake plugin:gettext[foreman_discovery]`) and got:

```
rake aborted!
NoMethodError: undefined method `match' for nil:NilClass
/home/orabin/.rvm/gems/ruby-2.2.3@foreman/gems/gettext_i18n_rails-1.4.0/lib/gettext_i18n_rails/tasks.rb:59:in `block (2 levels) in <top (required)>'
/home/orabin/foreman/lib/tasks/gettext.rake:43:in `block in <top (required)>'
Tasks: TOP => gettext:po:update => gettext:po:it:update => /home/orabin/git/foreman/foreman_discovery/locale/it/foreman_discovery.po => /home/orabin/git/foreman/foreman_discovery/locale/it/foreman_discovery.edit.po => gettext:internal:force
(See full trace by running task with --trace)
```

running it again worked.

### CODE REVIEW

File: locale/Makefile [hunk 1]
```diff
@@ -44,33 +41,20 @@ uniq-po:
...
-		--sort-by-msgid \
-		--no-wrap \
-		--no-location \
-		-o ${DOMAIN}.pot \
-		--package-name=${DOMAIN} \
-		--package-version="${VERSION}" \
-		--msgid-bugs-address=foreman-dev@googlegroups.com \
-		--copyright-holder="Foreman developers" \
-		--copyright-year=$(shell date +%Y) \
-		$(shell find ../app -type f -name *.rb -o -name *.erb)
```
Comment (lzap) [inline locale/Makefile:76]: Can you keep this target for the future? It looks like the rake task is unstable, I'd like to keep it handy when it does not work so release is not blocked. You can rename it to `external-extract` or something like that, maybe add a coment why we have it here.

**Comment by lzap** [comment]
LGTM otherwise, no fuzzy translations...

**Comment by lzap** [comment]
One more thing, when I run it (with the core PR), I see `foreman.po` files created in locale dir:

```
## orrabin-8874...orrabin/8874
?? locale/ca/foreman.po
?? locale/de/foreman.po
?? locale/en_GB/foreman.po
?? locale/es/foreman.po
?? locale/foreman.pot
?? locale/fr/foreman.po
?? locale/gl/foreman.po
?? locale/it/foreman.po
?? locale/ja/foreman.po
?? locale/ko/foreman.po
?? locale/pt_BR/foreman.po
?? locale/ru/foreman.po
?? locale/sv_SE/foreman.po
?? locale/zh_CN/foreman.po
?? locale/zh_TW/foreman.po
```

Do you see the same?

**Comment by orrabin** [comment]
@lzap no I only see foreman_discovery.po files but there was a bug about this before. What steps are you doing?

**Comment by lzap** [comment]
Ok that's fine. Please see my comment and I can merge.

**Comment by orrabin** [comment]
@lzap I put back the extract in the makefile. Tests aren't running yet because `test_plugin_foreman_discovery_develop` is failing.

**Comment by lzap** [comment]
That's because https://github.com/theforeman/foreman/pull/3082/files#diff-6afdff346c7e417ea7f74f437487706a perhaps, but this is not code change, I will merge anyway. Thanks.

**Comment by lzap** [comment]
Merged as 23922c2740c35e262cd5d2df3a6606a02433c848, thank you!

**Comment by domcleal** [comment]
Don't forget to update the MO files too.  Looks like you might have missed step 3 of http://projects.theforeman.org/projects/foreman/wiki/How_to_Create_a_Plugin#Pulling-translations-from-Transifex.

**Pull request closed without merging**